### PR TITLE
Provides the initial release of common.js and some helpers in Tagliatelle.

### PR DESCRIPTION
**Pull request by andyHa**
common.js and ie-polyfill.js are destined to be our new core JS libraries where we aggregate helpers and utilities currently spread across all products.

The additionally provided helpers in Tagliatelle simplify combining several files and also improve caching in dev environments.

### CODE REVIEW

File: src/main/java/sirius/tagliatelle/compiler/IncludeHandler.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        if (!Tagliatelle.isAssignableTo(expression.getType(), String.class)) {
+            compiler.getContext().error(position, "A constant String is required as parameter for include!");
+            return ConstantEmitter.EMPTY;
+        }
+
+        String resourcePath = (String) expression.eval(null);
+
+        if (!resourcePath.startsWith("/assets") && !resourcePath.startsWith("assets/")) {
```
Comment (qw3ry) [inline src/main/java/sirius/tagliatelle/compiler/IncludeHandler.java:70]:
```suggestion
        if (!resourcePath.startsWith("/assets/") && !resourcePath.startsWith("assets/")) {
```

**Review by qw3ry — APPROVED** [review]

File: src/main/resources/default/assets/common/polyfill.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            k++;
+        }
+        return false;
+    };
+}
+/**
+ * Polyfill for the String.startsWith function
+ * see https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith#Polyfill
+ */
+if (!String.prototype.startsWith) {
```
Comment (sabieber) [inline src/main/resources/default/assets/common/polyfill.js:53]: s.o.

File: src/main/resources/default/assets/common/polyfill.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        position = position || 0;
+        return this.indexOf(searchString, position) === position;
+    };
+}
+
+/**
+ * Polyfill for the String.endsWith function
+ * see https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith
+ */
+if (!String.prototype.endsWith) {
```
Comment (sabieber) [inline src/main/resources/default/assets/common/polyfill.js:64]: s.o.

**Review by sabieber — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/tagliatelle/macros/InlineResourceMacro.java [hunk 4]
```diff
@@ -67,6 +76,6 @@ public String getName() {
 
     @Override
     public String getDescription() {
-        return "Returns the contents of the given asset as string.";
+        return "Returns the contents of the given asset as string. Use escaceJS() to place the data in a JS string.";
```
Comment (jakobvogel) [inline src/main/java/sirius/tagliatelle/macros/InlineResourceMacro.java:79]:
```suggestion
        return "Returns the contents of the given asset as string. Use escapeJS() to place the data in a JS string.";
```

**Review by sabieber — APPROVED** [review]

File: src/main/resources/default/assets/common/common-polyfills.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            if (searchElement === currentElement ||
+                (searchElement !== searchElement && currentElement !== currentElement)) { // NaN !== NaN
+                return true;
+            }
+            k++;
+        }
+        return false;
+    };
+}
+/**
```
Comment (jakobvogel) [inline src/main/resources/default/assets/common/common-polyfills.js:49]:
```suggestion
}

/**
```

**Review by jakobvogel — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**